src/desktop/DesktopNativeCryptoFacade.ts

```diff
@@ -1,49 +1,69 @@
 
 type FsExports = typeof FsModule
 
 export class DesktopNativeCryptoFacade implements NativeCryptoFacade {
-	constructor(private readonly fs: FsExports, private readonly cryptoFns: CryptoFunctions, private readonly utils: DesktopUtils) {}
+	constructor(private readonly fs: FsExports, private readonly cryptoFns: CryptoFunctions, private readonly tfs: TempFs) {}
 
 	aesEncryptObject(encryptionKey: Aes256Key, object: number | string | boolean | ReadonlyArray<any> | {}): string {
 		const serializedObject = JSON.stringify(object)
 		const encryptedBytes = this.cryptoFns.aesEncrypt(encryptionKey, stringToUtf8Uint8Array(serializedObject))
 		return uint8ArrayToBase64(encryptedBytes)
 	}
 
-	aesDecryptObject(encryptionKey: Aes256Key, serializedObject: string): {} {
+	aesDecryptObject(encryptionKey: Aes256Key, serializedObject: string): Record<string, unknown> {
 		const encryptedBytes = base64ToUint8Array(serializedObject)
 		const decryptedBytes = this.cryptoFns.aesDecrypt(encryptionKey, encryptedBytes, true)
 		const stringObject = utf8Uint8ArrayToString(decryptedBytes)
 		return JSON.parse(stringObject)
 	}
 
+	async aesEncryptFile(key: Uint8Array, fileUri: string): Promise<EncryptedFileInfo> {
+		// at the moment, this is randomized if the file to be encrypted
+		// was created with FileFacade.writeDataFile.
+		// to make it safe in all conditions, we should re-generate a random file name.
+		// we're also not checking if the file to be encrypted is actually located in
+		// the temp scratch space
+		const bytes = await this.fs.promises.readFile(fileUri)
+		const keyBits = this.cryptoFns.bytesToKey(key)
+		const encrypted = this.cryptoFns.aesEncrypt(keyBits, bytes)
+		const targetDir = await this.tfs.ensureEncryptedDir()
+		const writtenFileName = path.basename(fileUri)
+		const filePath = path.join(targetDir, writtenFileName)
+		await this.fs.promises.writeFile(filePath, encrypted)
+		return {
+			uri: filePath,
+			unencryptedSize: bytes.length,
+		}
+	}
+
 	/**
 	 * decrypts a file and returns the decrypted files path
 	 */
 	async aesDecryptFile(key: Uint8Array, encryptedFileUri: FileUri): Promise<FileUri> {
-		const targetDir = path.join(this.utils.getTutanotaTempPath(), "decrypted")
-		await this.fs.promises.mkdir(targetDir, { recursive: true })
-
+		const targetDir = await this.tfs.ensureUnencrytpedDir()
 		const encData = await this.fs.promises.readFile(encryptedFileUri)
 		const bitKey = this.cryptoFns.bytesToKey(key)
 		const decData = this.cryptoFns.aesDecrypt(bitKey, encData, true)
 
 		const filesInDirectory = await this.fs.promises.readdir(targetDir)
-		const newFilename = nonClobberingFilename(filesInDirectory, path.basename(encryptedFileUri))
+		// since we're working purely in scratch space until putFileIntoDownloadsFolder
+		// is called, we could re-generate a random name here.
+		const writtenFileName = path.basename(encryptedFileUri)
+		const newFilename = nonClobberingFilename(filesInDirectory, writtenFileName)
```

Comment:
with a random name we wouldn't need all this clobbering filenames I guess?

Reply:
we only fixed the problem for generated attachment uploads for now, this is also called in a context where we download actual attachments and don´t do randomization yet